daemon/api_notices.go

```diff
@@ -213,3 +221,30 @@ func noticeViewableByUser(notice *state.Notice, requestUID uint32) bool {
 	}
 	return requestUID == userID
 }
+
+type noticeInfo struct {
+	ID            string            `json:"id"`
+	UserID        *uint32           `json:"user-id"`
+	Type          string            `json:"type"`
+	Key           string            `json:"key"`
+	FirstOccurred time.Time         `json:"first-occurred"`
+	LastOccurred  time.Time         `json:"last-occurred"`
+	LastRepeated  time.Time         `json:"last-repeated"`
+	Occurrences   int               `json:"occurrences"`
+	LastData      map[string]string `json:"last-data,omitempty"`
+	RepeatAfter   string            `json:"repeat-after,omitempty"`
+	ExpireAfter   string            `json:"expire-after,omitempty"`
+}
+
+func notice2noticeInfo(n *state.Notice) *noticeInfo {
+	raw, err := json.Marshal(n)
+	if err != nil {
+		logger.Panicf("internal error: cannot marshal notice: %v", err)
+	}
+	var info noticeInfo
+	err = json.Unmarshal(raw, &info)
```

Comment:
this is a strange way to do this for production code, I would follow the way change2changeInfo does this